### ISSUE TITLE
Categories are being pulled from other Articulate blogs

### ISSUE THREAD

**Issue by goverdhang** [Shazwazza/Articulate#160]
Hello Shazwazza - I am using Articulate 1.0.9 and ran into an issue. I had to setup several blogs as part of my architecture and when I try to render the categories and display on a page, the categories from all other blogs are getting retrieved. This issue was mentioned earlier here ( #26 , #75 ) but, I don't see a fix yet. 

   Can we expect a fix for this soon? Thank You!!

**Comment by Shazwazza** [comment]
Both of those issues are 'fixed' (closed) and you are using an old version, you should upgrade to the latest version and test. You will require Umbraco 7.3 as a minimum

**Comment by ukdiverboi** [comment]
You say this has been resolved in the latest version, however I have checked the latest 2.5 source code and can see the following method and related content.

```
        public static IEnumerable<string> GetAllCategories(this UmbracoHelper helper, IMasterModel masterModel)
        {
            //TODO: Make this somehow only lookup tag categories that are relavent to posts underneath the current Articulate root node!
            return helper.TagQuery.GetAllContentTags("ArticulateCategories").Select(x => x.Text).OrderBy(x => x);
        }
```

This is still returning all categories from both articulate root nodes in the tree. Am I missing something??

**Comment by Shazwazza** [comment]
Did you test with the latest version?

I'll have to look through the code, it's been a while since I've looked at this - my advice is to test with the latest version and verify if it works or doesn't work (regardless of what comments you find in the code)

**Comment by yalamber** [comment]
Hello Shazwazza,
I did test this with latest version. 
Umbarco: 7.4.2 and Articulate 2.0.5
But I have the same issue. Any pointer on how to fix this would be helpful. 
Thank you

**Comment by yalamber** [comment]
I had it working by modifying GetAllCategories method with similar to how tags were working and it worked ok.

**Comment by Shazwazza** [comment]
Have pushed a fix for this now with notes about fixing Umbraco core to better support this. Will be available in next major release

**Comment by LexMiranda** [comment]
yalamber, can you tell me what did you exactly changed in GetAllCategories?

**Comment by WiseyD** [comment]
I'm having this issue with Articluate 2.05 on Umbraco 7.5.4.

**Comment by Shazwazza** [comment]
The fix is listed for version 3.0.0, you'll need to upgrade
